sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -385,9 +385,16 @@ object DateTimeUtils {
    * microseconds where microsecond 0 is 1970-01-01 00:00:00Z.
    */
   def instantToMicros(instant: Instant): Long = {
-    val us = Math.multiplyExact(instant.getEpochSecond, MICROS_PER_SECOND)
-    val result = Math.addExact(us, NANOSECONDS.toMicros(instant.getNano))
-    result
+    val secs = instant.getEpochSecond
+    if (secs == Math.floorDiv(Long.MinValue, MICROS_PER_SECOND)) {
```

Comment:
Could you introduce a constant for `Math.floorDiv(Long.MinValue, MICROS_PER_SECOND)`.

Reply:
Ok, I will